### PR TITLE
2273 Added webhook-sentry proxy to send webhooks securely

### PR DESCRIPTION
**Pull request by albertisfu**
I've added the configuration to use `webhook-sentry` as a proxy when sending webhook events.

- Added `juggernaut/webhook-sentry:latest` image to our docker-compose so it's available in dev and testing.
- Added some tests to confirm the internal IPs are blocked
- Increased the request timeout to 3 seconds since the proxy would add some additional delay, using 1 second I was getting some timeout errors.

I found an incompatibility using the proxy and `ngrok`, the request is received but it arrives without content, I'll dig in a bit more to see if is a problem when using any proxy and `ngrok` or something specific to `webhook-sentry.`

Now to test webhooks are sent properly through the proxy I'm using: https://webhook.site/ it generates a URL where you can check the webhook request.

To test protection against SSRF you can use:
`https://make-127-0-0-1-rr.1u.ms`

### CODE REVIEW

File: cl/settings/project/security.py [hunk 1]
```diff
@@ -9,6 +9,10 @@
     "ALLOWED_HOSTS", default=["www.courtlistener.com"]
 )
 
+WEBHOOKS_PROXY_HOST = env(
+    "WEBHOOKS_PROXY_HOST", default="http://cl-webhook-sentry:9090"
+)
```
Comment (mlissner) [inline cl/settings/project/security.py:14]: I think in the future there's a good chance we'll start using this for other things too, like scraping. Better to call it `EGRESS_PROXY_HOST` instead?

File: cl/api/tests.py [hunk 2]
```diff
@@ -968,3 +971,72 @@ def test_lacks_access(self) -> None:
...
+    def test_avoid_sending_webhooks_to_internal_ips(self):
+        """Can we avoid sending webhooks to internal IPs?"""
+
+        webhook_event = WebhookEventFactory(
+            webhook=self.webhook_https,
+            content="{'message': 'ok_1'}",
+            event_status=WEBHOOK_EVENT_STATUS.IN_PROGRESS,
+        )
+        send_webhook_event(webhook_event)
+        webhook_to_compare = WebhookEvent.objects.get(pk=webhook_event.pk)
```
Comment (mlissner) [inline cl/api/tests.py:1011]: Can't you just do `webhook_event.refresh_from_db()`?

**Review by mlissner — CHANGES_REQUESTED** [review]
Looks good. One small thing, but we can't merge until we've got the darned thing deployed in prod. Gotta figure that out.

File: cl/api/tests.py (hunk 2 above)
Comment (albertisfu) [reply cl/api/tests.py]: yeah, this work well here, thanks!

File: cl/settings/project/security.py (hunk 1 above)
Comment (albertisfu) [reply cl/settings/project/security.py]: Changed!

**Comment by albertisfu** [comment]
Great, I've done the requested changes.
Thanks!

**Comment by mlissner** [comment]
Cool. We'll merge once we've got the proxy in prod. Thank you.

**Pull request merged**